contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,286 @@
+    }
+
+    function depositStake(uint256 stake) internal {
+        uint256 senderBalance = balances[msg.sender];
+
+        if (msg.value == stake) {
+            return;
+        }
+
+        if (msg.value < stake) {
+            require(senderBalance >= stake.sub(msg.value), "Insufficient balance");
+            balances[msg.sender] = senderBalance.sub(stake.sub(msg.value));
+        } else {
+            balances[msg.sender] = senderBalance.add(msg.value.sub(stake));
+        }
+
+        Deposit(msg.sender, msg.value);
+    }
+
+    function playerOneReveal(bytes32 gameKey, bytes32 secret, uint8 move) external whenNotPaused {
+
+        require(move < 3, INVALID_MOVE_MSG);
+        Game storage game = games[gameKey];
+        uint256 stake = game.stake;
+        require(stake != 0, GAME_NOT_FOUND_MSG);
+        require(game.step == Steps.PLAYER_TWO_MOVE, INVALID_STEP_MSG);
+        address playerOne = game.playerOne;
+        require(msg.sender == playerOne, ONLY_PLAYER_ONE_CALL_MSG);
+
+
+        //Validate move
+        bytes32 expectedMoveHash = createGameKey(secret, move);
+        require(gameKey == expectedMoveHash, "Move and secret do not match");
+
+        game.playerOneMove = move;
+        game.step = Steps.PLAYER_ONE_REVEAL;
+    }
+
+    /**
+    * Once the opponent has made their move, the creator of the game must resolve to determine the winner
+    * and reward the winner.
+    * If the creator does not resolve the game then the opponent can collect their winnings
+     * after the forfeit window has expired.
+    */
+    function playerTwoReveal(bytes32 gameKey, bytes32 secret, uint8 playerTwoMove) external whenNotPaused {
+
+        Game storage game = games[gameKey];
+        uint256 stake = game.stake;
+        require(game.step == Steps.PLAYER_ONE_REVEAL, INVALID_STEP_MSG);
+        address playerTwo = game.playerTwo;
+        require(msg.sender == playerTwo, INVALID_PLAYER_MSG);
+        require(playerTwoMove < 3, INVALID_MOVE_MSG);
+
+        //Validate move
+        bytes32 expectedMoveHash = createOpponentMoveHash(gameKey, secret, playerTwoMove);
+        require(game.playerTwoMoveHash == expectedMoveHash, "Move and secret do not match");
+
+        uint8 playerOneMove = game.playerOneMove;
+
+
+        //Draw
+        if(playerOneMove == playerTwoMove){
+            resetGame(game);
+            balances[playerOne] = balances[playerOne].add(stake);
+            balances[playerTwo] = balances[playerTwo].add(stake);
+            emit Draw(gameKey);
+            return;
+        }
+
+        //Determine winner
+        address winner;
+        address loser;
+        uint256 winnings = SafeMath.mul(stake, 2);
+
+        // If the creator wins then is rewarded by not paying a fee on the winnings
+        if (isWinner(playerOneMove, opponentMove)) {
+            winner = playerOne;
+            loser = playerTwo;
+        } else {
+            winner = playerTwo;
+            loser = playerTwo;
```

Comment:
What about using `loser` too?

Reply:
I see that bug of making  playerTwo the loser in both instances! Yet to write my tests!

I do this further down:

`balances[loser] = balances[loser].add(fee);`